src/app/pages/account/users/store/user.effects.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { Store } from '@ngrx/store';
+import { TranslateService } from '@ngx-translate/core';
+import { of } from 'rxjs';
+import {
+  catchError, filter, map, switchMap,
+} from 'rxjs/operators';
+import { ApiEventMessage } from 'app/enums/api-event-message.enum';
+import { QueryParams } from 'app/interfaces/query-api.interface';
+import { User } from 'app/interfaces/user.interface';
+import {
+  userPageEntered, usersLoaded, usersNotLoaded, userAdded, userChanged, userRemoved,
+} from 'app/pages/account/users/store/user.actions';
+import { WebSocketService } from 'app/services';
+import { AppState } from 'app/store';
+import { builtinUsersToggled } from 'app/store/preferences/preferences.actions';
+import { waitForPreferences } from 'app/store/preferences/preferences.selectors';
+
+@Injectable()
+export class UserEffects {
+  loadUsers$ = createEffect(() => this.actions$.pipe(
+    ofType(userPageEntered, builtinUsersToggled),
+    switchMap(() => this.store$.pipe(waitForPreferences)),
+    switchMap((preferences) => {
+      let params: QueryParams<User> = [];
+      if (preferences.hideBuiltinUsers) {
+        // TODO: Fix any
+        params = [[['OR', [['builtin', '=', false], ['username', '=', 'root']]]]] as any;
```

Comment:
I'm sorry about that.
If you know how to fix type checking to make this expression valid please let me know.
![image](https://user-images.githubusercontent.com/351613/157479659-6aacd36f-9f76-4666-aa02-125a31495759.png)

Reply:
From standpoint of typings, there are too many braces.
```
[['OR', [['builtin', '=', false], ['username', '=', 'root']]]];
```

Reply:
> From standpoint of typings, there are too many braces.
> 
> ```
> [['OR', [['builtin', '=', false], ['username', '=', 'root']]]];
> ```

Yeah, it passes typescript validation but middleware throwing invalid syntax error in this case

Reply:
Then there is an issue in typings, either in QueryParams or ApiDirectory. You can try to fix it or forcecast with `as unknown as QueryParams<User>` leaving TODO comment.